rootless-install.sh

```diff
@@ -0,0 +1,330 @@
+For example:
+echo \"$(id -un):100000:65536\" >> /etc/subuid"
+		exit 1
+	fi
+	if ! grep "^$(id -un):\|^$(id -u):" /etc/subgid >/dev/null 2>&1; then
+		>&2 echo "Could not find records for the current user $(id -un) from /etc/subgid . Please make sure valid subuid range is set there.
+For example:
+echo \"$(id -un):100000:65536\" >> /etc/subgid"
+		exit 1
+	fi
+}
+
+start_docker() {
+	# detect if overlay is supported (ubuntu)
+	tmpdir=$(mktemp -d)
+	mkdir -p $tmpdir/lower $tmpdir/upper $tmpdir/work $tmpdir/merged
+	if "$BIN"/rootlesskit mount -t overlay overlay -olowerdir=$tmpdir/lower,upperdir=$tmpdir/upper,workdir=$tmpdir/work $tmpdir/merged >/dev/null 2>&1; then
+		USE_OVERLAY=1
+	fi
+	rm -rf "$tmpdir"
+
+	if [ -z "$SYSTEMD" ]; then
+		start_docker_nonsystemd
+		return
+	fi
+
+	mkdir -p $HOME/.config/systemd/user
+	
+	DOCKERD_FLAGS="--experimental"
+	
+	if [ -n "$SKIP_IPTABLES" ]; then
+		DOCKERD_FLAGS="$DOCKERD_FLAGS --iptables=false"
+	fi
+
+	if [ "$USE_OVERLAY" = "1" ]; then
+		DOCKERD_FLAGS="$DOCKERD_FLAGS --storage-driver=overlay"
+	else
+		DOCKERD_FLAGS="$DOCKERD_FLAGS --storage-driver=vfs"
+	fi
+
+	CFG_DIR="$HOME/.config"
+	if [ -n "$XDG_CONFIG_HOME" ]; then
+		CFG_DIR="$XDG_CONFIG_HOME"
+	fi
+
+
+	if [ ! -f $CFG_DIR/systemd/user/docker.service ]; then
+		cat <<EOT > $CFG_DIR/systemd/user/docker.service
+[Unit]
+Description=Docker Application Container Engine (Rootless)
+Documentation=https://docs.docker.com
+
+[Service]
+Environment=PATH=$HOME/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+ExecStart=$HOME/bin/dockerd-rootless.sh $DOCKERD_FLAGS
+ExecReload=/bin/kill -s HUP \$MAINPID
+TimeoutSec=0
+RestartSec=2
+Restart=always
+StartLimitBurst=3
+StartLimitInterval=60s
+LimitNOFILE=infinity
+LimitNPROC=infinity
+LimitCORE=infinity
+TasksMax=infinity
+Delegate=yes
+Type=simple
+
+[Install]
+WantedBy=multi-user.target
+EOT
+	systemctl --user daemon-reload
+	fi
+	if ! systemctl --user status docker >/dev/null 2>&1; then
+		echo "# starting systemd service"
+		systemctl --user start docker
+	fi
+	systemctl --user status docker | cat
+
+	sleep 1
+	PATH="$BIN:$PATH" DOCKER_HOST="unix://$XDG_RUNTIME_DIR/docker.sock" docker version
+}
+
+service_instructions() {
+	if [ -z "$SYSTEMD" ]; then
+		return
+	fi
+	cat <<EOT
+#
+# To control docker service run:
+# systemctl --user (start|stop|restart) docker
+#
+EOT
+}
+
+
+start_docker_nonsystemd() {
+	iptablesflag=
+	if [ -n "$SKIP_IPTABLES" ]; then
+		iptablesflag="--iptables=false "
+	fi
+	cat <<EOT
+# systemd not detected, dockerd daemon needs to be started manually
+
+$BIN/dockerd-rootless.sh --experimental $iptablesflag--storage-driver vfs
+
+EOT
+}
+
+print_instructions() {
+	start_docker
+	echo "# Docker binaries are installed in $BIN"
+	if [ "$(which $DAEMON)" != "$BIN/$DAEMON" ]; then
+		echo "# WARN: dockerd is not in your current PATH or pointing to $BIN/$DAEMON"
+	fi
+	echo "# Please make sure following environment variables are set (or put them to ~/.bashrc):\n"
```

Comment:
Minor nit (just saw it in Andrew's output) (re https://github.com/docker/cli/issues/770)

```suggestion
	echo "# Make sure following environment variables are set (or put them to ~/.bashrc):\n"
```